src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java

```diff
@@ -11,33 +12,31 @@
 @EnableWebSecurity
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
 
-  /**
-   * Stuff which should not need authentication.
-   */
-  @Override
-  public void configure(WebSecurity web) throws Exception {
-    web.ignoring().antMatchers("/static/**").antMatchers("/favicon.ico");
-  }
+    /**
+     * Stuff which should not need authentication.
+     */
+    @Override
+    public void configure(WebSecurity web) throws Exception {
+        web.ignoring().antMatchers("/static/**").antMatchers("/favicon.ico");
+    }
 
-  @Override
-  protected void configure(HttpSecurity http) throws Exception {
-//    http
-//      .authorizeRequests()
-//      .antMatchers(HttpMethod.POST, "/event/**")
-//      .fullyAuthenticated();
-    
-    http
-      .authorizeRequests()
-      .antMatchers(HttpMethod.DELETE, "/**")
-      .denyAll();
-  
-    http
-      .antMatcher("/**")
-      .authorizeRequests()
-      .anyRequest().permitAll();
-    
-    // For REST calls (need more investigation)
-    http.csrf().disable();
-  }
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.httpBasic().and().authorizeRequests()
+                .antMatchers(HttpMethod.GET,"/event/**").permitAll()
+                .antMatchers("/event/**")
+                .hasRole("ADMIN")
+                .and()
+                .antMatcher("/**")
+                .authorizeRequests()
+                .anyRequest().permitAll();
+        http.csrf().disable();
+        http.headers().frameOptions().disable();
```

Comment:
Becasue of : https://spring.io/blog/2013/08/23/spring-security-3-2-0-rc1-highlights-security-headers/#x-frame-options